src/txn/txn_recover.c

```diff
@@ -700,6 +700,9 @@ __recovery_setup_file(WT_RECOVERY *r, const char *uri, const char *config)
     WT_RET(__wt_config_getones(r->session, config, "id", &cval));
     fileid = (uint32_t)cval.val;
 
+    /* We just want the ID, regardless of the namespace it's in. */
+    FLD_CLR(fileid, WT_BTREE_ID_NAMESPACE_SHARED);
```

Comment:
This would work, but would it be possible to future-proof this part of code (like if we add any more namespace types in the future) by clearing the top three bits of the ID here?